gradle-jdks/src/main/java/com/palantir/gradle/jdks/ToolchainsPlugin.java

```
@@ -0,0 +1,162 @@
+        if (!JdksPlugin.getEnableGradleJdkProperty(rootProject)) {
+            throw new RuntimeException("Cannot apply ToolchainsJdksPlugin without enabling gradle.jdk.setup.enabled");
+        }
+
+        rootProject
+                .getLogger()
+                .info("Gradle JDK automanagement is enabled. The JDKs used for all subprojects "
+                        + "are managed by the configured custom toolchains.");
+        JdkDistributions jdkDistributions = new JdkDistributions();
+
+        JdksExtension jdksExtension = JdksPlugin.extension(rootProject, jdkDistributions);
+
+        rootProject.getPluginManager().withPlugin("com.palantir.baseline-java-versions", unused -> {
+            rootProject
+                    .getExtensions()
+                    .getByType(BaselineJavaVersionsExtension.class)
+                    .getSetupJdkToolchains()
+                    .set(false);
+        });
+        TaskProvider<Wrapper> wrapperTask = rootProject.getTasks().named("wrapper", Wrapper.class);
+
+        TaskProvider<GenerateGradleJdkConfigsTask> generateGradleJdkConfigs = rootProject
+                .getTasks()
+                .register("generateGradleJdkConfigs", GenerateGradleJdkConfigsTask.class, task -> {
+                    task.getDaemonJavaVersion().set(jdksExtension.getDaemonTarget());
+                    task.getJavaVersionToJdkDistros()
+                            .putAll(rootProject.provider(() -> JdkDistributionConfigurator.getJavaVersionToJdkDistros(
+                                    rootProject,
+                                    jdkDistributions,
+                                    jdksExtension.getConfiguredJavaVersions().get().stream()
+                                            .filter(javaLanguageVersion ->
+                                                    javaLanguageVersion.canCompileOrRun(MINIMUM_SUPPORTED_JAVA_VERSION))
+                                            .collect(Collectors.toSet()),
```

I've thought about it a bit more, and it seems a bit odd to be able to support using jdk 8 but not be able run gradle with it. If I've understood correctly, this is just the version of the jvm used to run the gradle daemon right? If that's the case, then perhaps we fail explicitly on the daemon java version being set to 8 so that we can't get into a bad state.

---

The failure will happen during the configuration phase: https://github.com/palantir/gradle-jdks/blob/develop/gradle-jdks-setup/src/main/resources/gradle-jdks-setup.sh#L154. Basically if I add to the `jdkExtension` java versions < 8, when we will run the line: 
`"$java_home"/bin/java -cp "$APP_GRADLE_DIR"/gradle-jdks-setup.jar com.palantir.gradle.jdks.setup.GradleJdkInstallationSetup` with the `java_home` == version 8, it will fail with `Unsupported class version` because `GradleJdkInstallationSetup` was compiled with java version 11, while the current java version would be 8.

I think I could refactor the script such that this line runs with a different java home (probably use the gradle daemon java version to do the setup for all the jdks that I am downloading). This will involve some refactoring though.